### ISSUE TITLE
Index error when creating new column

### ISSUE THREAD

**Issue by mar-muel** [modin-project/modin#651]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04
- **Modin installed from (source or binary)**: pip install git+https://github.com/modin-project/modin
- **Modin version**: 0.5.2
- **Python version**: 3.6.7
- **Exact command to reproduce**:
```python
import modin.pandas as pd

pd.DataFrame({'col1': ['test'], 'col2': ['Hello world']}).to_csv('test.csv', index=False)
df = pd.read_csv('test.csv')
df['col3'] = 'new column value'
print(df)
```

Raises the following error:
```
Traceback (most recent call last):
  File "test.py", line 12, in <module>
    print(df)
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/pandas/base.py", line 3263, in __str__
    return repr(self)
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/pandas/dataframe.py", line 93, in __repr__
    result = repr(self._build_repr_df(num_rows, num_cols))
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/pandas/base.py", line 61, in _build_repr_df
    head_front = head.to_pandas()
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/backends/pandas/query_compiler.py", line 479, in to_pandas
    df = self.data.to_pandas(is_transposed=self._is_transposed)
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/engines/base/frame/partition_manager.py", line 463, in to_pandas
    [obj.to_pandas() for obj in part] for part in self.partitions
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/engines/base/frame/partition_manager.py", line 463, in <listcomp>
    [obj.to_pandas() for obj in part] for part in self.partitions
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/engines/base/frame/partition_manager.py", line 463, in <listcomp>
    [obj.to_pandas() for obj in part] for part in self.partitions
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/engines/ray/pandas_on_ray/frame/partition.py", line 87, in to_pandas
    dataframe = self.get()
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/engines/ray/pandas_on_ray/frame/partition.py", line 34, in get
    handle_ray_task_error(e)
  File "/home/mar-muel/anaconda3/envs/myenv/lib/python3.6/site-packages/modin/engines/ray/utils.py", line 13, in handle_ray_task_error
    raise getattr(builtins, s.split(":")[0])("".join(s.split(":")[1:]))
IndexError:  index 0 is out of bounds for axis 0 with size 0
```

Weirdly enough, this doesn't happen when the dataframe only consists of 1 column:
```python
pd.DataFrame({'col1': ['test']}).to_csv('test.csv', index=False)
df = pd.read_csv('test.csv')
df['col2'] = 'new column value'
print(df)
```
Output:
```
   col1              col2
0  test  new column value
```
No idea what's going on here. Everything works fine when using vanilla pandas or when not reading from CSV. Thanks for your help!

**Comment by devin-petersohn** [comment]
Thanks @mar-muel for the report!

I can reproduce this, it doesn't happen in debug mode `MODIN_DEBUG=1 ipython`, so it's not a metadata issue. I also can confirm that the insert is happening. I will take a look at this and get this fixed. Thanks again for the report!

**Comment by williamma12** [comment]
Hi @mar-muel, this should be fixed when #663 was merged in and it works for me on the current master. Can you try reinstalling modin with
```
pip install git+https://github.com/modin-project/modin
```
and see if it works for you?

**Comment by devin-petersohn** [comment]
@mar-muel I am going to close this as we were able to reproduce and fix it. Please reopen if it still is not working on master for you.
